jme3-xr/src/main/java/com/jme3/system/lwjgl/openxr/HelloOpenXRGL.java

```diff
@@ -0,0 +1,886 @@
+                    "The OpenXR runtime supports only OpenGL " + maxMajorVersion + "." + minMajorVersion +
+                    " and earlier, but we got OpenGL " + actualMajorVersion + "." + actualMinorVersion
+                );
+            }
+            logger.info("OnCreateXr: Win=" + window + " Egl=" + useEglGraphicsBinding);
+            
+            //Bind the OpenGL context to the OpenXR instance and create the session
+            PointerBuffer pp = stack.mallocPointer(1);
+            check(xrCreateSession(
+                xrInstance,
+                XRHelper.createGraphicsBindingOpenGL(
+                    XrSessionCreateInfo.malloc(stack)
+                        .type$Default()
+                        .next(NULL)
+                        .createFlags(0)
+                        .systemId(systemID),
+                    stack,
+                    window,
+                    useEglGraphicsBinding
+                ),
+                pp
+            ));
+
+            xrSession = new XrSession(pp.get(0), xrInstance);
+
+            if (!missingXrDebug && !useEglGraphicsBinding) {
+                XrDebugUtilsMessengerCreateInfoEXT ciDebugUtils = XrDebugUtilsMessengerCreateInfoEXT.calloc(stack)
+                    .type$Default()
+                    .messageSeverities(
+                        XR_DEBUG_UTILS_MESSAGE_SEVERITY_INFO_BIT_EXT |
+                        XR_DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT |
+                        XR_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT
+                    )
+                    .messageTypes(
+                        XR_DEBUG_UTILS_MESSAGE_TYPE_GENERAL_BIT_EXT |
+                        XR_DEBUG_UTILS_MESSAGE_TYPE_VALIDATION_BIT_EXT |
+                        XR_DEBUG_UTILS_MESSAGE_TYPE_PERFORMANCE_BIT_EXT |
+                        XR_DEBUG_UTILS_MESSAGE_TYPE_CONFORMANCE_BIT_EXT
+                    )
+                    .userCallback((messageSeverity, messageTypes, pCallbackData, userData) -> {
+                        XrDebugUtilsMessengerCallbackDataEXT callbackData = XrDebugUtilsMessengerCallbackDataEXT.create(pCallbackData);
+                        System.out.println("XR Debug Utils: " + callbackData.messageString());
+                        return 0;
+                    });
+
+                System.out.println("Enabling OpenXR debug utils");
+                check(xrCreateDebugUtilsMessengerEXT(xrInstance, ciDebugUtils, pp));
+                xrDebugMessenger = new XrDebugUtilsMessengerEXT(pp.get(0), xrInstance);
+            }
+        }
+    }
+
+    public void createXRReferenceSpace() {
+        try (MemoryStack stack = stackPush()) {
+            PointerBuffer pp = stack.mallocPointer(1);
+
+            check(xrCreateReferenceSpace(
+                xrSession,
+                XrReferenceSpaceCreateInfo.malloc(stack)
+                    .type$Default()
+                    .next(NULL)
+                    .referenceSpaceType(XR_REFERENCE_SPACE_TYPE_LOCAL)
```

Comment:
Should this be XR_REFERENCE_SPACE_TYPE_LOCAL or XR_REFERENCE_SPACE_TYPE_STAGE. I found that it was surprising that my face was in the floor when I tried this; because XR_REFERENCE_SPACE_TYPE_LOCAL puts the origin at the headset position when the application is started but XR_REFERENCE_SPACE_TYPE_STAGE puts it at the origin of the play area (usually at floor level). Or maybe we should let the applications choose.

I thought this was a useful explanation


> The difference between XR_REFERENCE_SPACE_TYPE_STAGE and XR_REFERENCE_SPACE_TYPE_LOCAL in OpenXR comes down to the origin and orientation of the coordinate system that the application will use to represent the position and orientation of devices, like headsets and controllers, within the virtual world.
> 
> In XR_REFERENCE_SPACE_TYPE_LOCAL, the origin is typically located at the position where the XR system was initialized. For example, in a seated VR experience, this could be the location of the headset at the time the application was started. The Y-axis is typically up, and the orientation is typically such that the forward direction (negative Z-axis) is facing in the initial direction of the user.
> 
> XR_REFERENCE_SPACE_TYPE_STAGE, on the other hand, is designed for standing, room-scale experiences. The origin is typically located at the center of the user's designated play area (if one has been defined), with the Y-axis pointing up, and the forward direction (negative Z-axis) is often facing a user-defined forward direction in the play area (like the direction facing the computer or the front of the room).
> 
> The practical differences a user might experience would depend on the application, but here are some examples:
> 
> In a room-scale VR game, if XR_REFERENCE_SPACE_TYPE_LOCAL is used, the user's initial position in the game might not be in the center of the play area, but somewhere near the edge or corner, because the origin is where the system was initialized, not the center of the play area. If the user moves around in the play area, they might move out of the game's designated area. Using XR_REFERENCE_SPACE_TYPE_STAGE in this case would be better, because it ensures the user starts in the center of the game's area and can freely move around within it.
> 
> If the user is sitting at a desk and not moving around (like in a cockpit simulator game), the difference between XR_REFERENCE_SPACE_TYPE_LOCAL and XR_REFERENCE_SPACE_TYPE_STAGE might not be noticeable, because the user is not using the play area. But if XR_REFERENCE_SPACE_TYPE_STAGE is used and the user's chair is not in the center of the play area, they might find themselves starting off-center in the game, which could be disorienting.
> 
> In general, the choice between XR_REFERENCE_SPACE_TYPE_LOCAL and XR_REFERENCE_SPACE_TYPE_STAGE should be based on the type of experience the application provides (seated vs. room-scale) and the user's physical environment.
>

Reply:
At least it should be configurable for app-developers.
I tried it with XR_REFERENCE_SPACE_TYPE_STAGE, and still moving does not work.
Just the rotation starts initial in wrong direction.